internal/assets/manager.go

```diff
@@ -168,25 +173,147 @@ func (am *assetManager) GetTokenPools(ctx context.Context, ns string, typeName s
 	return am.database.GetTokenPools(ctx, am.scopeNS(ns, filter))
 }
 
-func (am *assetManager) GetTokenPool(ctx context.Context, ns, typeName, name string) (*fftypes.TokenPool, error) {
+func (am *assetManager) GetTokenPool(ctx context.Context, ns, typeName, poolName string) (*fftypes.TokenPool, error) {
 	if _, err := am.selectTokenPlugin(ctx, typeName); err != nil {
 		return nil, err
 	}
 	if err := fftypes.ValidateFFNameField(ctx, ns, "namespace"); err != nil {
 		return nil, err
 	}
-	if err := fftypes.ValidateFFNameField(ctx, name, "name"); err != nil {
+	if err := fftypes.ValidateFFNameField(ctx, poolName, "name"); err != nil {
 		return nil, err
 	}
-	return am.database.GetTokenPool(ctx, ns, name)
+	pool, err := am.database.GetTokenPool(ctx, ns, poolName)
+	if err != nil {
+		return nil, err
+	}
+	if pool == nil {
+		return nil, i18n.NewError(ctx, i18n.Msg404NotFound)
+	}
+	return pool, nil
 }
 
-func (am *assetManager) GetTokenAccounts(ctx context.Context, ns, typeName, name string, filter database.AndFilter) ([]*fftypes.TokenAccount, *database.FilterResult, error) {
+func (am *assetManager) GetTokenAccounts(ctx context.Context, ns, typeName, poolName string, filter database.AndFilter) ([]*fftypes.TokenAccount, *database.FilterResult, error) {
+	pool, err := am.GetTokenPool(ctx, ns, typeName, poolName)
+	if err != nil {
+		return nil, nil, err
+	}
+	return am.database.GetTokenAccounts(ctx, filter.Condition(filter.Builder().Eq("poolprotocolid", pool.ProtocolID)))
+}
+
+func (am *assetManager) GetTokenTransfers(ctx context.Context, ns, typeName, name string, filter database.AndFilter) ([]*fftypes.TokenTransfer, *database.FilterResult, error) {
 	pool, err := am.GetTokenPool(ctx, ns, typeName, name)
 	if err != nil {
 		return nil, nil, err
 	}
-	return am.database.GetTokenAccounts(ctx, filter.Condition(filter.Builder().Eq("protocolid", pool.ProtocolID)))
+	return am.database.GetTokenTransfers(ctx, filter.Condition(filter.Builder().Eq("poolprotocolid", pool.ProtocolID)))
+}
+
+func (am *assetManager) MintTokens(ctx context.Context, ns, typeName, poolName string, transfer *fftypes.TokenTransfer, waitConfirm bool) (*fftypes.TokenTransfer, error) {
+	transfer.Type = fftypes.TokenTransferTypeMint
+	if transfer.Author == "" {
+		transfer.Author = config.GetString(config.OrgIdentity)
+	}
+	transfer.From = ""
+	if transfer.To == "" {
+		transfer.To = transfer.Author
+	}
+	return am.transferTokensWithID(ctx, fftypes.NewUUID(), ns, typeName, poolName, transfer, waitConfirm)
+}
+
+func (am *assetManager) BurnTokens(ctx context.Context, ns, typeName, poolName string, transfer *fftypes.TokenTransfer, waitConfirm bool) (*fftypes.TokenTransfer, error) {
+	transfer.Type = fftypes.TokenTransferTypeBurn
+	if transfer.Author == "" {
+		transfer.Author = config.GetString(config.OrgIdentity)
+	}
+	if transfer.From == "" {
+		transfer.From = transfer.Author
```

Comment:
Again want to be sure this is `Author` here, rather than `Key`.
My understanding was the tokens features weren't going to _require_ any kind of registration of identity, so would support raw keys only. The `Author` field would be used consistently where if it's resolvable to a registered key, that would work... but we'd also support indexing transactions where we don't have an author.

Holding on the rest of the review until you've had a chance to think about this feedback, as I think it might be significant.